core/src/main/scala/medeia/decoder/ErrorStack.scala

```diff
@@ -0,0 +1,21 @@
+package medeia.decoder
+
+final case class ErrorStack(frames: List[StackFrame]) {
+  def push(frame: StackFrame): ErrorStack = ErrorStack(frame +: frames)
```

Comment:
why not `frame :: frames`

Reply:
Personal preference :shrug:  `+:` is the generalized version and `::` is the special case for `List`.  I normally stick to `+:`, because that would *in theory* :joy: make it easier to just swap the list for e.g. a `Vector` without replacing all `::` :shrug:

Reply:
:shrug: fine with me :)